### PR TITLE
feat: ipfs comparisons

### LINKED ISSUE #779: Page of "How is IPFS different from..."
* bittorrent
* storj
* arweave
* sia
* filecoin
* dat
* ???

### PR DESCRIPTION
**Pull request by DannyS03**
## Context
- Document on comparing IPFS to other [similar] storage solutions
- Incorporates some suggestions in the linked issue and utilizes a table to compare common parameters and features
- Unsure where the best place is for this to live; under case studies atm
- resolves #779

## Latest preview

Please view the latest Fleek preview [here](https://bafybeiece5dct2obrlaxebta2k7mi7fcqd2777e6gazxisb3smzzyjbyfy.on.fleek.co/concepts/comparisons/).

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

File: docs/concepts/comparisons.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: 'IPFS Comparisons'
+description: Comparing IPFS to other solutions.
+---
+
+# Storage Solution Comparisons
+
+## How is IPFS different from other solutions?
+
+IPFS is a general-purpose file system that uses a distributed hash table (DHT) to store and retrieve data. This sets it apart from other solutions with a more specific focus or using different data storage mechanisms. For example:
```
Comment (ElPaisano) [inline docs/concepts/comparisons.md:10]: Suggestion: This sentence 
```
This sets it apart from other solutions with a more specific focus or using different data storage mechanisms. For example:
```
reads a little strange. I think rephrasing it to something like
```
This sets it apart from other storage solutions that use different data storage mechanisms or have a more specific focus .  Similar solutions include:
```
would read better.

File: docs/concepts/comparisons.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  > Filecoin is built on IPFS and uses the IPFS network for data storage and retrieval. Filecoin and IPFS are complementary technologies providing decentralized and efficient storage solutions.
+- [Dat](https://datproject.org/) is a decentralized data-sharing tool that uses a distributed hash table (DHT) for data storage. It focuses on enabling data sharing and collaboration.
+- [Holo](https://holochain.org/) is a decentralized hosting platform that uses a unique data storage and sharing mechanism called Holochain. It allows users to host and run web-based applications on a peer-to-peer network.
+- [Swarm](https://swarm-gateways.net/bzz:/theswarm.eth/) is a decentralized storage and sharing platform built on the Ethereum blockchain. It uses smart contracts and cryptographic techniques to securely store and share data. It focuses on providing a decentralized, secure, and censorship-resistant storage solution.
+
+## Comparing the key features of other solutions to IPFS
+
+The following table outlines key features of storage mechanisms and how they compare to IPFS.
+> Note that all these solutions use content-based addressing.
```
Comment (ElPaisano) [inline docs/concepts/comparisons.md:24]: Two comments:
- Phrasing suggestion "Note that all of these solutions use content-based addressing."
- Might be better to use a callout box here:
   ```
   :::callout
   Note that all of these solutions use content-based addressing. 
   ::::
   ```

File: docs/concepts/comparisons.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+- [BitTorrent](https://www.bittorrent.com/) is a peer-to-peer (P2P) file-sharing protocol that uses a centralized tracker to manage the distribution of files among peers. It focuses on file-sharing rather than file storage.
+- [Storj](https://storj.io/) and [Sia](https://sia.tech/) are decentralized cloud storage platforms that use distributed networks of nodes for data storage. They focus on providing cloud storage services rather than a general-purpose distributed file system.
+- [Arweave](https://www.arweave.org/) is a decentralized, permanent storage platform that uses a novel data structure called a "blockweave" for data storage. It focuses on providing permanent storage rather than a file-sharing system.
+- [Filecoin](https://filecoin.io/) is a decentralized storage network that allows users to rent out disk space. It focuses on providing a decentralized storage marketplace. It uses a proof-of-replication consensus mechanism and supports payment in various cryptocurrencies.
+  > Filecoin is built on IPFS and uses the IPFS network for data storage and retrieval. Filecoin and IPFS are complementary technologies providing decentralized and efficient storage solutions.
```
Comment (ElPaisano) [inline docs/concepts/comparisons.md:16]: Would suggest a `:::callout` here

**Review by ElPaisano — CHANGES_REQUESTED** [review]
Hi @DannyS03 I like the idea of adding this info into the docs. 

Made a few review notes

RE location in the docs: I'm not sure how I feel about this page being in Case Studies. The way I understand a Case Studies section, it's a section of the docs where examples of applications / other things that use / integrate / build on the software (in this case, IPFS), so this page feels out of place in this section, since it's a comparison of IPFS to other technologies. I'd actually suggest adding it as a subsection on the /concepts/what-is-ipfs/ page instead or at least somewhere in the IPFS 101 subsection under Concepts. Open to thoughts here

File: docs/.vuepress/config.js [hunk 4]
```diff
@@ -269,6 +269,7 @@ module.exports = {
           {
             title: 'Case studies',
             children: [
+              ['/concepts/comparisons', 'Comparisons'],
```
Comment (2color) [inline docs/.vuepress/config.js:272]: This isn't a case study. I would consider putting this in the IPFS 101 category.

File: docs/concepts/comparisons.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- [Dat](https://datproject.org/) is a decentralized data-sharing tool that uses a distributed hash table (DHT) for data storage. It focuses on enabling data sharing and collaboration.
```
Comment (2color) [inline docs/concepts/comparisons.md:17]: It's more commonly referred to as the Hypercore protocol nowadays. https://hypercore-protocol.org/

**Review by 2color — CHANGES_REQUESTED** [review]
I would recommend making a clear distinction between two main categories here:
- Crypto-economic networks like Filecoin and Arweave
- Protocols made for moving (routing & transferring) bytes around like BitTorrent, Hypercore, IPFS

I'd also split those into two different tables to avoid creating a misleading comparison between the two categories

Some illuminating resources: 
https://blog.mauve.moe/posts/protocol-comparisons
https://medium.com/decentralized-web/comparing-ipfs-and-dat-8f3891d3a603

File: docs/.vuepress/config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ElPaisano) [reply docs/.vuepress/config.js:272]: > I'd actually suggest adding it as a subsection on the /concepts/what-is-ipfs/ page instead or at least somewhere in the IPFS 101 subsection under Concepts. 

Seems like there's a shared consensus here @DannyS03. Can you move this out to IPFS 101?

File: docs/concepts/comparisons.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+---
+title: 'IPFS Comparisons'
+description: Comparing IPFS to other solutions.
+---
+
+# Storage Solution Comparisons
```
[the remark this answers is not in the document]
Comment (ElPaisano) [reply docs/concepts/comparisons.md:6]: @2color created this issue https://github.com/ipfs/ipfs-docs/issues/1368 based on your callout. I also added your notion to the docs reorg notion

[the remark this answers is not in the document]
Comment (2color) [reply docs/concepts/comparisons.md:6]: Thanks, @ElPaisano. That's great

**Comment by 2color** [comment]
Btw, I should add that this page is very valuable. Might be worth also adding Git, which is also a distributed content-addressed system that has many parallels to IPFS.

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Review by ElPaisano — APPROVED** [review]

**Comment by ElPaisano** [comment]
@DannyS03 changes look good 👍 @2color looks like DannyS03 addressed your changes. Is this good to go from your end? If so, we can merge once you approve.

**Review by 2color — APPROVED** [review]

**Comment by DannyS03** [comment]
Thanks for the suggestions @ElPaisano @2color! To Daniel's point, let's continue keeping an eye out for this page and consider other solutions + comparisons.

**Pull request merged**